### ISSUE TITLE
RFE : uncover-like to give gaps in article mode

### ISSUE THREAD

**Issue by ysalmon** [josephwright/beamer#498]
Greetings,
I am using beamer and beamerarticle to typeset course material : I give students the article version at the beginning of the hour and they are supposed to add their own notes to it while we discuss the material with the presentation as a visual support.

At some point there is a part of a sentence that I want to uncover on the presentation after discussion, and of course I also do not want it to appear on the article version : I want the printed text to have a gap for the students to fill.

But while ``uncover`` works nicely in presentation mode, it does not produce the wanted gap in article mode (as per Beamer documentation, which states that ``uncover`` behaves like ``only`` in that mode).

    \documentclass{article}
    \usepackage{beamerarticle}
    \begin{document}
    \begin{frame}
    toto\uncover<presentation:2- | article:0>{tata}titi
    \end{frame}
    \end{document}

[A question on tex.sx][1] deals with handout mode, but the accepted solution does not work _mutatis mutandis_ in article mode.

I understand the rationale for ``uncover`` to behave this way in article mode : most often, one will want to get the full material in the article version. Since there are no slides in article mode, the ``article:0`` would be a non ambiguous way to tell beamerarticle otherwise.

Meanwhile, is there a way to somehow coerce beamerarticle ?

  [1]: https://tex.stackexchange.com/questions/225077/producing-lecture-notes-and-handouts-with-spaces-for-students-to-fill-in-details

**Comment by louisstuart96** [comment]
In presentation mode, `\uncover` and `\invisible` are involved with "transparency", which requires `pgf` modules as well as the generation of multi-slide frames. In article mode there is no chance (and no need) to produce such frames, thus `pgf` modules are not loaded, and these commands are processed simply.

For your case, a temporary solution could be:
```
\documentclass{article}
\usepackage{beamerarticle}

\makeatletter
\mode
<all>
{%
  \newcommand<>\placehold[1]{%
    \setbox\@tempboxa\hbox{#1}%
    % Use the below definition if color goes wrong
    % \setbox\@tempboxa\hbox{\color@begingroup#1\color@endgroup}%
    \alt#2{\box\@tempboxa}{\hbox to \wd\@tempboxa{\hfill}}%  
  }%
}
\makeatother

\begin{document}
\begin{frame}
toto\placehold<presentation:2- | article:0>{tata}titi
\end{frame}
\end{document}
```

**Comment by ysalmon** [comment]
Thank you so much !

**Comment by josephwright** [comment]
I'm not sure this should be in `beamer`: the intention of `\uncover` and so on is for presentation effects ...

**Comment by ysalmon** [comment]
I do not know if it is a faux-pas to comment on my old closed RFE, but while I happily used @louisstuart96 ``\placehold`` for some time, I recently stumbled upon the difficulty of hiding a text that linebreaks, which of course clashes with it being typeset in a ``\hbox``.

So I investigated a little in beamer source and tried to replicate the behaviour of ``\visible`` in article mode. _Miraculously_ and in spite of my very low understanding of pgf etc., it seems to work. Not daring to propose a PR here, I just posted it [on TeX.SX](https://tex.stackexchange.com/a/475006/103608).
